### PR TITLE
fix(styles): horizon progress indicator

### PR DESCRIPTION
**Pull request by g-cheishvili**
## Related Issue

Part of #3354

## Description
Added Horizon theming support as good as it was possible. A lot of diff, had to hack way around to make things work.

### Before:
image
image

### After:
image
image

#### Please check whether the PR fulfills the following requirements

1. The output matches the design specs
- [x] All values are in `rem`
- [x] Text elements follow the truncation rules
- [x] hover state of the element follow design spec
- [x] focus state of the element follow design spec
- [x] active state of the element follow design spec
- [x] selected state of the element follow design spec
- [x] selected hover state of the element follow design spec
- [x] pressed state of the element follow design spec
- [x] Responsiveness rules - the component has modifier classes for all breakpoints
- [x] Includes Compact/Cosy/Tablet design
- [x] RTL support
2. The code follows fundamental-styles code standards and style
- [x] only one top level `fd-*` class is used in the file
- [x] BEM naming convention is used
- [x] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [x] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [x] `fd-reset()` mixin is applied to all elements
- [x] Variables are used, if some value is used more than twice.
- [x] Checked if current components can be reused, instead of having new code.
3. Testing
- [x] tested Storybook examples with "CSS Resources" `normalize` option 
- [x] tested Storybook examples with "CSS Resources" `unnormalize` option 
- [ n/a ] Verified all styles in IE11
- [ n/a ] Updated tests
- [x] last commit message should have `[ci visual]` so it can trigger chromatic visual regression (e.g. `test: run chromatic visual regression [ci visual]`)
4. Documentation
- [x] Storybook documentation has been created/updated
- [x] Breaking Changes [wiki](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes) has been updated in case of breaking changes.

[Breaking Changes](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes#progress-indicator-3429)

### CODE REVIEW

**Review by N1XUS — COMMENTED** [review]
please add [ci visual] to last commit message

File: src/styles/progress-indicator.scss [hunk 1]
```diff
@@ -77,18 +158,47 @@ $fd-progress-indicator-states: (
...
     @include fd-reset();
     @include fd-flex();
-    @include fd-ellipsis();
+
+    @include fd-progress-indicator-popover-control() {
+      flex-grow: 1;
+    }
+  }
+
+  &__icon {
```
[the remark this answers is not in the document]
Comment (g-cheishvili) [reply src/styles/progress-indicator.scss:167]: I have to move both label and icon in different directions when Horizon is applied, so I had to extract them

**Review by platon-rov — COMMENTED** [review]
* Fiori, No gap before icon

Screenshot 2022-05-19 at 14 06 00

* Fiori, HC, Icons not centered vertically

image

**Comment by g-cheishvili** [comment]
> * Fiori, No gap before icon
> 
> Screenshot 2022-05-19 at 14 06 00
> 
> * Fiori, HC, Icons not centered vertically
> 
> image

Fixed again. Now it's according to specs

**Review by platon-rov — APPROVED** [review]

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
In fiori mode, this rating indicator overflow text appears to have lost its ellipses: 
Screen Shot 2022-05-24 at 8 49 01 AM

File: src/styles/progress-indicator.scss [hunk 2]
```diff
@@ -2,93 +2,232 @@
...
+      border-radius: 50%;
+      top: 0.0005rem;
+    }
+
+    &::before {
+      left: 0.0005rem;
+    }
+
+    &::after {
+      right: 0.0005rem;
```
Comment (mikerodonnell89) [inline src/styles/progress-indicator.scss:125]: The specs call for 0.063rem here, giving 1px between the dots and the edge of the indicator

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
The top one is missing the white outline around the progress bar
Screen Shot 2022-05-24 at 9 35 04 AM

File: src/styles/progress-indicator.scss (hunk 2 above)
Comment (g-cheishvili) [reply src/styles/progress-indicator.scss:125]: Yes, you are right, that should be 0, because border makes that 1px spacing

**Pull request merged**